src/main/java/net/hypixel/nerdbot/command/GeneratorCommands.java

```diff
@@ -668,4 +691,21 @@ private void addCommandToUserHistory(User user, String command) {
             discordUserRepository.findById(user.getId()).getGeneratorHistory().addCommand(command);
         }
     }
-}
+
+    /**
+     * Gets the gen command auto hide preference from a {@link GuildSlashEvent}.
+     *
+     * @param event The {@link GuildSlashEvent} triggered by the user you want to get the auto hide preference from.
+     *
+     * @return The auto hide preference from the user.
+     */
+    private boolean getUserAutoHideSetting(GuildSlashEvent event) {
+        DiscordUserRepository repository = NerdBotApp.getBot().getDatabase().getRepositoryManager().getRepository(DiscordUserRepository.class);
+        DiscordUser user = repository.findById(event.getMember().getId());
+        if (user != null) {
+            return user.isAutoHideGenCommands();
+        }
+        return AUTO_HIDE_ON_ERROR;
+
+    }
```

Comment:
```suggestion
    private boolean getUserAutoHideSetting(GuildSlashEvent event) {
        DiscordUserRepository repository = NerdBotApp.getBot().getDatabase().getRepositoryManager().getRepository(DiscordUserRepository.class);
        DiscordUser user = repository.findById(event.getMember().getId());
        
        if (user != null) {
            return user.isAutoHideGenCommands();
        }
        
        return AUTO_HIDE_ON_ERROR;
    }
```

Reply:
Even though I didn't commit the suggestion it seems to already be this way.